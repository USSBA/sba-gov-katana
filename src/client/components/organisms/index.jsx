/* IMPORTS */

// miscellaenous organisms
import BusinessGuideTileCollection from "./business-guide-tile-collection/business-guide-tile-collection.jsx";
import CardCollection from "./card-collection/card-collection.jsx";
import DeveloperTester from "./developer-tester/developer-tester.jsx";
import DocumentCardCollection from "./document-card-collection/document-card-collection.jsx";
import FrontPageHero from "./front-page-hero/front-page-hero.jsx";
import FundingProgramsTileCollection from "./funding-programs-tile-collection/funding-programs-tile-collection.jsx";
import DisasterAlert from "./header-footer/disaster-alert/disaster-alert.jsx";
import NotificationBar from "./header-footer/notification-bar.jsx";
import Hero from "./hero/hero.jsx";
import RelatedDocumentCards from "./related-document-cards/related-document-cards.jsx";
import SearchBox from "./search-box/search-box.jsx";
import SectionNav from "./section-nav/section-nav.jsx";

// header-footer organisms
import Footer from "./header-footer/footer/footer.jsx";
import MobileNav from "./header-footer/mobile-nav/mobile-nav.jsx";
import Header from "./header-footer/header/header.jsx";

// homepage organisms
import Blog from "./homepage/blog/blog.jsx";
import FrontPageLady from "./homepage/front-page-lady/front-page-lady.jsx";
import HappeningNow from "./homepage/happening-now/happening-now.jsx";

// lender match organisms
// -- form
import AdditionalInfoForm from "./lender-match/form/additional-info.jsx";
import BusinessInfoForm from "./lender-match/form/business-info.jsx";
import ContactInfoForm from "./lender-match/form/contact-info.jsx";
import IndustryInfoForm from "./lender-match/form/industry-info.jsx";
import LoanInfo from "./lender-match/form/loan-info.jsx";
import ReviewSection from "./lender-match/form/review-page-helpers.jsx";
import ReviewSubmitInfoForm from "./lender-match/form/review-submit-info.jsx";
// -- landing page
import FindLendersIntro from "./lender-match/landing-page/find-lenders-intro.jsx";
import HelpfulQuestions from "./lender-match/landing-page/helpful-questions.jsx";
import {HowItWorksSection} from "./lender-match/landing-page/how-it-works.jsx";
import PreparationChecklist from "./lender-match/landing-page/preparation-checklist.jsx";
// -- success page
import ConfirmSection from "./lender-match/success-page/confirm-section.jsx";
import DynamicCounselingAndTools from "./lender-match/success-page/counseling-and-tools.jsx";
import CounselorMap from "./lender-match/success-page/counselor-map.jsx";
import EmailConfirmationInvalid from "./lender-match/success-page/email-confirmed-page.jsx";

// modal organisms
import LeaveSbaModal from "./modals/leave-sba-modal/leave-sba-modal.jsx";
import MobileSectionNavModal from "./modals/mobile-section-nav/mobile-section-nav.jsx";

/* EXPORTS */

// miscellaneous organisms
export {
	BusinessGuideTileCollection,
	CardCollection,
	DeveloperTester,
	DocumentCardCollection,
	FrontPageHero,
	FundingProgramsTileCollection,
	DisasterAlert,
	NotificationBar,
	Hero,
	RelatedDocumentCards,
	SearchBox,
	SectionNav
};

// header-footer organisms
export {
	Footer,
	MobileNav,
	Header
};

// homepage organisms
export {
	Blog,
	FrontPageLady,
	HappeningNow
};

// lender match organisms
export {
	AdditionalInfoForm,
	BusinessInfoForm,
	ContactInfoForm,
	IndustryInfoForm,
	LoanInfo,
	ReviewSection,
	ReviewSubmitInfoForm,
	FindLendersIntro,
	HelpfulQuestions,
	HowItWorksSection,
	PreparationChecklist,
	ConfirmSection,
	DynamicCounselingAndTools,
	CounselorMap,
	EmailConfirmationInvalid
};

// modal organisms
export {
	LeaveSbaModal,
	MobileSectionNavModal
};
