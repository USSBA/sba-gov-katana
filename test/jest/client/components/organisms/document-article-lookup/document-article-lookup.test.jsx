/*global expect*/

import React from 'react'
import renderer from 'react-test-renderer'
import _ from 'lodash'
import { shallow } from 'enzyme'

import { DocumentArticleLookup } from 'organisms'

const submit = jest.fn()
const reset = jest.fn()
const queryChange = jest.fn()
const pageChange = jest.fn()

let lookupProps = {
  title: 'This is a title',
  queryState: {},
  items: [
    {
      name: 'Obi-Wan Kenobi',
      title: 'Master',
      lightsaberColor: 'Green',
      age: '40',
      type: 'jedi'
    },
    {
      name: 'Luke Skywalker',
      title: 'Knight',
      lightsaberColor: 'Blue',
      age: '20',
      type: 'jedi'
    },
    {
      name: 'Yoda',
      title: 'Master',
      lightsaberColor: 'Green',
      age: '900',
      type: 'jedi'
    },
    {
      name: 'Bastilla Shan',
      title: 'Knight',
      lightsaberColor: 'Yellow',
      age: '24',
      type: 'jedi'
    }
  ],
  itemCount: 42,
  pageNumber: 2,
  pageSize: 4,
  taxonomies: [
    {
      name: 'Title',
      terms: ['Padawan', 'Knight', 'Master']
    },
    {
      name: 'LightsaberColor',
      terms: ['Red', 'Green', 'Blue', 'Yellow']
    }
  ],
  onSubmit: submit,
  onReset: reset,
  onQueryChange: queryChange,
  onPageChange: pageChange,
  isFetching: false,
  fieldsToShowInDetails: ['Name', 'LightsaberColor'],
  type: 'jedi'
}
describe('DocumentArticleLookup', () => {
  test('should render with all the information', () => {
    let props = _.clone(lookupProps)
    const component = renderer.create(<DocumentArticleLookup {...props} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should fire the submit when the apply button is pressed ', () => {
    let props = _.clone(lookupProps)
    const component = shallow(<DocumentArticleLookup {...props} />)
    component.find('Button').simulate('click')
    expect(submit).toBeCalled()
  })
  //
  // test('should show important data properly ', () => {
  //   let props = _.clone(lookupProps);
  //   const component = shallow(<DocumentArticleLookup {...props}/>);
  //   expect(component.find("Paginator").prop("pageNumber")).toEqual(props.pageNumber);
  // });

  // test('should render without a zip code ', () => {
  //   let cardData = _.omit(card, "zipCode");
  //   const component = shallow(<LinkCard {...cardData}/>);
  //   expect(component.find(".document-article-lookup-citystatezip")).toHaveLength(0);
  // });
  //
  // test('should render without the link', () => {
  //   let cardData = _.omit(card, "link");
  //   const component = shallow(<LinkCard {...cardData}/>);
  //   expect(component.find(".document-article-lookup-link")).toHaveLength(0);
  // });
})
