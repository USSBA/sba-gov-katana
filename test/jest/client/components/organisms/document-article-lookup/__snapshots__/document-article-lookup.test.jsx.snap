// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentArticleLookup should render with all the information 1`] = `
<div>
  <div
    className="banner"
  >
    <h2
      className="header"
    >
      This is a title
    </h2>
    <div>
      <div
        className="searchBox"
      >
        <div
          className="inputContainer "
          hidden={undefined}
          id="document-lookup-text-input-container"
        >
          <label
            className="controlLabel"
            htmlFor="document-lookup-text-input"
          >
            Search
          </label>
          <div
            className="textInputContainer"
          >
            <input
              className="textInput "
              id="document-lookup-text-input"
              onChange={[Function]}
              onKeyUp={[Function]}
              placeholder="Search by title or number"
              value={undefined}
            />
          </div>
        </div>
        <div
          className="searchIcon"
        >
          <i
            className="search fa fa-search"
          />
        </div>
      </div>
      <div
        className="multiSelect"
      >
        <div
          id="title-select-container"
        >
          <label>
            Title
          </label>
          <div
            className="errorClass"
            id="title-select"
          >
            <div
              className="Select  myselect Select--single"
              style={undefined}
            >
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-2--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    aria-activedescendant="react-select-2--value"
                    aria-expanded={false}
                    aria-owns="react-select-2--value"
                    aria-readonly="false"
                    className="Select-input"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    style={
                      Object {
                        "border": 0,
                        "display": "inline-block",
                        "width": 1,
                      }
                    }
                    tabIndex={0}
                  />
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <img
                    alt="dropdown arrow"
                    className="chevronIcon"
                    src="test-file-stub"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="multiSelect"
      >
        <div
          id="lightsabercolor-select-container"
        >
          <label>
            Lightsaber Color
          </label>
          <div
            className="errorClass"
            id="lightsabercolor-select"
          >
            <div
              className="Select  myselect Select--single"
              style={undefined}
            >
              <div
                className="Select-control"
                onKeyDown={[Function]}
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                style={undefined}
              >
                <span
                  className="Select-multi-value-wrapper"
                  id="react-select-3--value"
                >
                  <div
                    className="Select-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    aria-activedescendant="react-select-3--value"
                    aria-expanded={false}
                    aria-owns="react-select-3--value"
                    aria-readonly="false"
                    className="Select-input"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    role="combobox"
                    style={
                      Object {
                        "border": 0,
                        "display": "inline-block",
                        "width": 1,
                      }
                    }
                    tabIndex={0}
                  />
                </span>
                <span
                  className="Select-arrow-zone"
                  onMouseDown={[Function]}
                >
                  <img
                    alt="dropdown arrow"
                    className="chevronIcon"
                    src="test-file-stub"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="applyButton"
      >
        <div
          id={null}
        >
          <button
            autoFocus={false}
            className="SmallInverseSecondaryButton applyButton "
            disabled={undefined}
            formTarget="_self"
            id={undefined}
            onKeyPress={[Function]}
            onTouchTap={[Function]}
            tabIndex="0"
          >
            Apply
          </button>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="paginator"
    >
      <div
        className="container paginator"
        id={undefined}
      >
        <p
          className="pagination-status"
        >
          <span>
            Showing  
          </span>
          <span
            className="bold"
          >
            5
              - 
            8
              
          </span>
          <span>
            of  
          </span>
          <span
            className="bold"
          >
            42
          </span>
        </p>
        <span
          className="previous-arrow"
        >
          <a
            className="ArrowButton "
            id={undefined}
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            <i
              alt="previous page"
              aria-hidden="true"
              className=" fa fa-chevron-left"
            />
          </a>
        </span>
        <span
          className="next-arrow"
        >
          <a
            className="ArrowButton noLeftBorder"
            id={undefined}
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            <i
              alt="next page"
              aria-hidden="true"
              className=" fa fa-chevron-right"
            />
          </a>
        </span>
      </div>
    </div>
    <div
      className="documentCardCollection"
    >
      <div
        className="document-card-collection cardCollection"
      >
        <div
          className="cardRow"
        >
          <div
            className="card-container card"
          >
            <div
              className="document-card-container "
            >
              <div>
                <div
                  className="documentTypeContainer"
                >
                  <div
                    className="documentType"
                  >
                    <div
                      className="type"
                    >
                      MEMO
                    </div>
                  </div>
                </div>
                <div />
                <a
                  autoFocus={false}
                  className=""
                  onKeyPress={[Function]}
                  onTouchTap={[Function]}
                  tabIndex="0"
                >
                  <h6
                    className="document-card-title title"
                  >
                    Master
                  </h6>
                </a>
                <div>
                  <div
                    className="dash"
                  >
                    <hr
                      className="decorativeDash"
                    />
                  </div>
                  <table>
                    <tbody />
                  </table>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card-container card"
          >
            <div
              className="document-card-container "
            >
              <div>
                <div
                  className="documentTypeContainer"
                >
                  <div
                    className="documentType"
                  >
                    <div
                      className="type"
                    >
                      MEMO
                    </div>
                  </div>
                </div>
                <div />
                <a
                  autoFocus={false}
                  className=""
                  onKeyPress={[Function]}
                  onTouchTap={[Function]}
                  tabIndex="0"
                >
                  <h6
                    className="document-card-title title"
                  >
                    Knight
                  </h6>
                </a>
                <div>
                  <div
                    className="dash"
                  >
                    <hr
                      className="decorativeDash"
                    />
                  </div>
                  <table>
                    <tbody />
                  </table>
                </div>
              </div>
            </div>
          </div>
          <div
            className="card-container card"
          >
            <div
              className="document-card-container "
            >
              <div>
                <div
                  className="documentTypeContainer"
                >
                  <div
                    className="documentType"
                  >
                    <div
                      className="type"
                    >
                      MEMO
                    </div>
                  </div>
                </div>
                <div />
                <a
                  autoFocus={false}
                  className=""
                  onKeyPress={[Function]}
                  onTouchTap={[Function]}
                  tabIndex="0"
                >
                  <h6
                    className="document-card-title title"
                  >
                    Master
                  </h6>
                </a>
                <div>
                  <div
                    className="dash"
                  >
                    <hr
                      className="decorativeDash"
                    />
                  </div>
                  <table>
                    <tbody />
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="cardRow"
        >
          <div
            className="card-container card"
          >
            <div
              className="document-card-container "
            >
              <div>
                <div
                  className="documentTypeContainer"
                >
                  <div
                    className="documentType"
                  >
                    <div
                      className="type"
                    >
                      MEMO
                    </div>
                  </div>
                </div>
                <div />
                <a
                  autoFocus={false}
                  className=""
                  onKeyPress={[Function]}
                  onTouchTap={[Function]}
                  tabIndex="0"
                >
                  <h6
                    className="document-card-title title"
                  >
                    Knight
                  </h6>
                </a>
                <div>
                  <div
                    className="dash"
                  >
                    <hr
                      className="decorativeDash"
                    />
                  </div>
                  <table>
                    <tbody />
                  </table>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="paginator"
    >
      <div
        className="container paginator"
        id={undefined}
      >
        <p
          className="pagination-status"
        >
          <span>
            Showing  
          </span>
          <span
            className="bold"
          >
            5
              - 
            8
              
          </span>
          <span>
            of  
          </span>
          <span
            className="bold"
          >
            42
          </span>
        </p>
        <span
          className="previous-arrow"
        >
          <a
            className="ArrowButton "
            id={undefined}
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            <i
              alt="previous page"
              aria-hidden="true"
              className=" fa fa-chevron-left"
            />
          </a>
        </span>
        <span
          className="next-arrow"
        >
          <a
            className="ArrowButton noLeftBorder"
            id={undefined}
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
          >
            <i
              alt="next page"
              aria-hidden="true"
              className=" fa fa-chevron-right"
            />
          </a>
        </span>
      </div>
    </div>
  </div>
</div>
`;
